roxie/ccd/ccdfile.cpp

```diff
@@ -1168,13 +1168,11 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
                 if ((dfsSize != (offset_t) -1 && dfsSize != f->getSize()) ||
                     (!dfsDate.isNull() && !dfsDate.equals(*f->queryDateTime(), false)))
                 {
-                    if (fileType == ROXIE_KEY)
-                    {
-                        // jhtree cache can keep files active and thus prevent us from loading a new version
+                    releaseSlaveDynamicFileCache();  // Slave dynamic file cache or...
```

Comment:
This will create a race condition with any concurrent call to querySlaveDynamicFileCache()